crates/ra_hir/src/ty/tests/data/array.txt

```diff
@@ -0,0 +1,52 @@
+[9; 10) 'x': &str
+[18; 19) 'y': isize
+[28; 293) '{     ... []; }': ()
+[38; 39) 'a': [_]
```

Comment:
the `_` means that we have a type variable here, which shouldn't ever happen -- all type variables should be replaced by their known type or `Ty::Unknown` after inference. The reason that this doesn't happen here is that `Ty::Array` is missing in `Ty::walk_mut`, which I missed so far. Can you add that there (should work exactly like the slice type)?